deb/ar.go

```diff
@@ -188,11 +201,23 @@ func parseArEntry(line []byte) (*ArEntry, error) {
 // like an `ar(1)` archive, and not some random file.
 func checkAr(reader io.Reader) error {
 	header := make([]byte, 8)
-	if _, err := reader.Read(header); err != nil {
-		return err
+	pos := 0
+	for pos < 8 {
+		count, err := reader.Read(header[pos:])
+		if  err == io.EOF {
+			if count+pos == 0  {
+				return fmt.Errorf("File is empty.")
+			} else {
+				return fmt.Errorf("Header too short for 'ar' file.")
+			}
+		}
+		if  err != nil {
+			return err
+		}
+		pos = pos + count
 	}
 	if string(header) != "!<arch>\n" {
-		return fmt.Errorf("Header doesn't look right!")
+		return fmt.Errorf("Header doesn't look as 'ar' file.")
```

Comment:
The english here is a bit broken - maybe `Header is invalid for an 'ar' archive.` instead?